packages/aws-client/src/KMS.ts

```diff
@@ -20,24 +23,26 @@ const createKey = (params = {}) => kms().createKey(params).promise();
  * @param {string} Plaintext - the string to be encrypted
  * @returns {Promise<string>} the Base 64 encoding of the encrypted value
  */
-const encrypt = (KeyId, Plaintext) =>
-  kms().encrypt({ KeyId, Plaintext }).promise()
-    .then(({ CiphertextBlob }) => CiphertextBlob.toString('base64'));
+export const encrypt = async (KeyId: string, Plaintext: string) => {
+  const { CiphertextBlob } = await kms().encrypt({ KeyId, Plaintext }).promise();
+
+  if (CiphertextBlob === undefined) throw new Error('Returned CiphertextBlob is undefined');
+
+  return CiphertextBlob.toString('base64');
+};
 
 /**
  * Decrypt a KMS-encrypted string, Base 64 encoded
  *
  * @param {string} ciphertext - a KMS-encrypted value, Base 64 encoded
  * @returns {string} the plaintext
  */
-const decryptBase64String = (ciphertext) =>
-  kms().decrypt({
+export const decryptBase64String = async (ciphertext: string) => {
+  const { Plaintext } = await kms().decrypt({
     CiphertextBlob: Buffer.from(ciphertext, 'base64')
-  }).promise()
-    .then(({ Plaintext }) => Plaintext.toString());
+  }).promise();
+
+  if (Plaintext === undefined) return undefined;
```

Comment:
So TS made you do this because it was warning you about calling `toString` on `undefined`?

Reply:
Correct, apparently `Plaintext` could potentially be undefined.